### DISCUSSION TITLE
TableExport Issue: 無法匯出PDF

### DISCUSSION THREAD

**Discussion by james-chao** [dotnetcore/BootstrapBlazor#3444] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

TableExport 在Excel File 部分可正常匯出，但無法正常匯出PDF file
BB 8.5.5 TableExport 8.1.0

### Expected Behavior

能正常匯出EXCEL 和 PDF file

### Steps To Reproduce

https://github.com/james-chao/FileViewerIssue.git

### .NET Version

8.0.204

### Anything else?

VS2022

**Comment by bb-auto** [comment]
@james-chao Thank you for reporting. We will give triage later.

**Reply by ArgoZhang** [reply]
@james-chao 这种问题不要第一时间就来发 `Issue` 正确解决问题办法

1. 去官网查看导出 PDF 是否正常
2. 如果官网不正常，创建 Issue
3. 如果官网正常，请查看相关文档

解决办法

https://www.blazor.zone/html2pdf

https://www.blazor.zone/export-pdf-button

增加包 `BootstrapBlazor.Html2Pdf`

请点击下方按钮 `Make as answer`

**Reply by james-chao** [reply]
@ArgoZhang 增加包 BootstrapBlazor.Html2Pdf 依然不行，請問單純設定 ShowExportButton="true" ShowExportPdfButton="true" ShowExportCsvButton="true" 匯出Csv和 Excel 都OK 只有 Pdf 不行，是因為Pdf 匯出還要參考 html2pdf 組件的寫法嗎?我在https://www.blazor.zone/table/export 上看导出 Pdf是可以的，但沒有特別寫Code啊，我的TableDemo 匯出如下 
EXCEL:
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/3213919/829ba92a-1b96-4c37-96df-108e56e04a6d)
PDF:
<img width="943" alt="image" src="https://github.com/dotnetcore/BootstrapBlazor/assets/3213919/b92ae836-0a0b-4eba-84d1-63ad3d387456">
<img width="956" alt="image" src="https://github.com/dotnetcore/BootstrapBlazor/assets/3213919/53892b0f-76a6-438f-9768-ccbf7308e93a">
https://github.com/james-chao/FileViewerIssue.git

**Reply by ArgoZhang** [reply, marked as answer]
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/22001478/77733dd1-137e-45f3-ab0b-f134c76f6802)


请点击下方按钮 Make as answer

**Reply by james-chao** [reply]
@ArgoZhang OK 了，感謝!

**Reply by ArgoZhang** [reply]
@james-chao 点一下按钮啊。不停的提醒，你也不点。

**Reply by james-chao** [reply]
@ArgoZhang 想再請問一下，我上到Server 後碰到以下的錯誤，也將builder.Services.Configure<HubOptions>(option => option.MaximumReceiveMessageSize = null); 設定了，call PuppeteerSharp 會出現超過30 秒的錯誤訊息，我要如何取消限制
類似await page.setDefaultNavigationTimeout(0); 這樣的設定，感謝!

    1) Exception Information
      ---------------------------------------------
      Exception Type: System.TimeoutException
      TargetSite: System.Threading.Tasks.Task <WithTimeout>b__0()
      Message: Timeout of 30000 ms exceeded
      Data: System.Collections.ListDictionaryInternal
      HelpLink: 
      Source: PuppeteerSharp
      HResult: -2146233083

**Reply by james-chao** [reply]
The issue was due to the fact that the server being deployed to did not have internet access and PuppeteerSharp was looking to download Chromium resolved by specifying a local execution path.

ExecutablePath = "C:\Program Files (x86)\Google\Chrome\Application\chrome.exe"

Revised code below

var url = ConfigurationManager.AppSettings["EnrolmentFormUrl"] + "?code=" + code;

var fileName = "enrolment" + $"{DateTime.Now:__ddMMyyyyhhmmssfff}" + ".pdf";

var output = Config.attachmentPath + fileName;                     

var browser = await Puppeteer.LaunchAsync(new LaunchOptions
{
    Headless = true,
    TransportFactory = AspNetWebSocketTransport.AspNetTransportFactory,
    ExecutablePath = "C:\Program Files (x86)\Google\Chrome\Application\chrome.exe"
}).ConfigureAwait(false);

var pdfOptions = new PdfOptions
{
    Format = PaperFormat.A4,
    PreferCSSPageSize = true
};

var page = await browser.NewPageAsync();
await page.GoToAsync(url).ConfigureAwait(false);
await page.PdfAsync(output, pdfOptions).ConfigureAwait(false);

**Reply by densen2014** [reply]
需要导出的组件id设置了吗，table外面再套一层div id=9527

**Reply by james-chao** [reply]
@densen2014 我在DEV 環境(local NB)是沒有問題的，PRD IIS server是不准上網的也沒有chrome,合理懷疑是無法下載Chromium 導致無法匯出PDF

**Comment by densen2014** [comment]
@ArgoZhang 是依赖chrome.exe吗?
